### PR TITLE
Upgrage libbpf, libbpfgo and go

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

I can build the main of libbpfgo and parca-agent locally, and both work using 1.20.1. I'm opening this PR to troubleshoot the CI issues.

1. Updates Go
2. Updates libbpf and libbpfgo
3. Updates golangci-lint
4. Update several pre-commit hooks
5. Changes upstream golangci-lint CI steps with a custom script to pass GO and CGO environment variables.

What fixes the CI failures:
```bash
GO_ENV := CGO_ENABLED=1 GOOS=linux GOARCH=$(ARCH) CC="$(CMD_CC)"
CGO_ENV := CGO_CFLAGS="$(CGO_CFLAGS)" CGO_LDFLAGS="$(CGO_LDFLAGS)"
$(GO_ENV) $(CGO_ENV) golangci-lint run
```

### Test 

* [x] Make all CI checks green
* [x] Manually run against Kubernetes locally
  * [x] Go
  * [x] C/C++ with .eh_frame

CleanShot 2023-03-03 at 16 10 55
CleanShot 2023-03-03 at 16 10 42
CleanShot 2023-03-03 at 16 03 55
CleanShot 2023-03-03 at 16 03 12
CleanShot 2023-03-03 at 16 03 25

fixes #1289 #1387

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -98,9 +97,12 @@ jobs:
       - name: Build
         run: make build
 
-      - name: Show kernel version
+      - name: Show Kernel version
```
Comment (javierhonduco) [inline .github/workflows/build.yml:100]: nit: Linux is capitalised as it's a proper noun, but a kernel is not

**Review by javierhonduco — APPROVED** [review]
Looks amazing!! ❤️  Could you please run some manual tests with Go stacks and a C/C++ application to ensure that the latest libbpf version doesn't have any obvious regression?

**Review by brancz — APPROVED** [review]
Lgtm if you’ve run the tests Javier described

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -27,12 +38,8 @@ repos:
           - --fuzzy-match-generates-todo
         types: [shell]
         exclude: ^internal/(?:go|pprof)/
-  - repo: https://github.com/golangci/golangci-lint
-    rev: v1.51.1
-    hooks:
-      - id: golangci-lint
```
Comment (maxbrunet) [inline .pre-commit-config.yaml:33]: I think overriding the `entry` of the existing hook should be enough and still ensure the correct version is available and used

```suggestion
  - repo: https://github.com/golangci/golangci-lint
    rev: v1.51.1
    hooks:
      - id: golangci-lint
        entry: make go/lint
```

Comment (kakkoyun) [reply .pre-commit-config.yaml:33]: I'll give this a try. Thanks for pointing that out ❤️

**Comment by javierhonduco** [comment]
Wohoo!! This is amazing!! Makes me so happy to see we now use the latest Go 🥳

**Pull request merged**